### ISSUE TITLE
changefeedccl: handle degraded (but not unavailable) sink

### ISSUE THREAD

**Issue by danhhz** [cockroachdb/cockroach#31200]
This is likely to require some tuning to our backpressure, but it's important for stability that we gracefully handle anything the external sink throws at us.

**Comment by tbg** [comment]
@danhhz is this the item you suggested could be addressed via toxiproxy testing but is likely not going to make the release, right?

**Comment by danhhz** [comment]
Yup. Writing the test is one part, and I'd be flabbergasted if it didn't involved a decent amount of work to actually make the test pass. This work would be awesome for hardening changefeed stability, but it's never really been in scope for 2.2 except as the most stretch of stretch goals. I had it on the list in case anyone thought it was interesting enough to pick up on a flex friday.
